src/main/java/mate/jdbc/dao/CarDaoImpl.java

```diff
@@ -0,0 +1,195 @@
+    public Optional<Car> get(Long id) {
+        String getCarById = "SELECT c.id as car_id, model, name, m.id as manufacturer_id, country "
+                + "FROM cars c JOIN manufacturers m ON c.manufacturer_id = m.id "
+                + "WHERE c.id = ? AND c.is_deleted = FALSE;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(getCarById)) {
+            statement.setLong(1, id);
+            statement.executeQuery();
+            ResultSet resultSet = statement.getResultSet();
+            Car car = null;
+            if (resultSet.next()) {
+                car = getCar(resultSet);
+            }
+            return Optional.ofNullable(car);
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get car from database by id: " + id, e);
+        }
+    }
+
+    @Override
+    public List<Car> getAll() {
+        String getAllCar = "SELECT c.id as car_id, model, name, m.id as manufacturer_id, country "
+                + "FROM cars c JOIN manufacturers m ON c.manufacturer_id = m.id "
+                + "WHERE c.is_deleted = FALSE;";
+        List<Car> carList = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(getAllCar)) {
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                carList.add(getCar(resultSet));
+            }
+            return carList;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get all cars from database.", e);
+        }
+    }
+
+    @Override
+    public Car update(Car car) {
+        String updateCar = "UPDATE cars "
+                + " SET model = ?, manufacturer_id = ? "
+                + "WHERE id = ? AND is_deleted = FALSE;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(updateCar)) {
+            statement.setString(1, car.getModel());
+            statement.setLong(2, car.getManufacturer().getId());
+            statement.setLong(3, car.getId());
+            statement.executeUpdate();
+            deleteDriversFromCar(car);
+            insertDriversToCar(car);
+            return car;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't update car in database: " + car, e);
+        }
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        String deleteCar = "UPDATE cars "
+                + "SET is_deleted = TRUE "
+                + "WHERE id = ?;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(deleteCar)) {
+            statement.setLong(1, id);
+            deleteDriversFromCar(get(id).orElseThrow());
```

Comment:
we shouldn't delete data from cars_drivers table, because we use soft-delete strategy